src/Network/Wai/SAML2.hs

```diff
@@ -208,7 +217,12 @@ data Result = Result {
     -- | An optional relay state, as provided in the POST request.
     relayState :: !(Maybe BS.ByteString),
     -- | The assertion obtained from the response that has been validated.
-    assertion :: !Assertion
+    assertion :: !Assertion,
+    -- | The ID of the request this result corresponds to to, if any. You
+    -- should check that it matches a request you generated
+    --
+    -- @since 0.4
+    inResponseTo :: !(Maybe T.Text)
```

Comment:
Similarly, to my comment for `Network.Wai.SAML2.Validation`, this could probably be the entire response, rather than just one field from it?

Reply:
Yes, I agree